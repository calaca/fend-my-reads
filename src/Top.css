.top {
  background-color: #f1f1f1;
  width: 100%;
  height: 80px;
  box-shadow: 0px 0px 15px 0px rgba(0, 0, 0, 0.15);
}

.top-title {
  color: #9c9c9c;
  text-align: center;
  font-size: 36px;
  font-weight: 400;
  text-transform: uppercase;
  margin: 0;
  padding-top: 20px;
}
