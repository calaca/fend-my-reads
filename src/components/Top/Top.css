.top {
  background-color: #f1f1f1;
  width: 100%;
  height: 80px;
  box-shadow: 0px 0px 15px 0px rgba(0, 0, 0, 0.15);
}

.top-title {
  color: #9c9c9c;
  text-align: center;
  font-size: 36px;
  font-weight: 400;
  text-transform: uppercase;
  margin: 0;
  padding-top: 20px;
}

.search-link {
  display: block;
  float: right;
  margin: -20px 30px 0 0;
  width: 70px;
  height: 70px;
  background-color: #d7d7d7;
  border-radius: 50%;
  box-shadow: 0px 0px 15px 0px rgba(0, 0, 0, 0.1);
  transition: all 0.4s ease-in-out;
}

.search-link:hover {
  background-color: #c9c9c9;
}

.search-link-icon {
  width: 38.12px;
  margin: 17px 0 0 16px;
}
