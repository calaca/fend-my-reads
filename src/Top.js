import React, { Component } from 'react';
import './Top.css';

class Top extends Component {
  render() {
    return (
      <div className="top">
        <h1 className="top-title">My Reads</h1>
      </div>
    )
  }
}

export default Top;
