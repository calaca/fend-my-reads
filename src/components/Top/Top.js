import React from 'react';
import { Link } from 'react-router-dom';
import searchIcon from '../../icons/search.svg';
import './Top.css';

const Top = () => {
  return (
    <div className="top">
      <h1 className="top-title">My Reads</h1>
      <Link className="search-link" to="/search">
        <img className="search-link-icon" src={searchIcon} alt="Search books" title="Search books"/>
      </Link>
    </div>
  )
}

export default Top;
