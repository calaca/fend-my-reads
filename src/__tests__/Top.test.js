import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Top from '../components/Top/Top';

describe('<Top />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Top />);
  });

  it('renders without crashing', () => {
    wrapper;
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the app name', () => {
    expect(wrapper.find('.top-title').contains('My Reads')).toBe(true);
  });

  it('has a Link to `/search`', () => {
    expect(wrapper.find('.search-link').prop('to')).toEqual('/search');
  });

  it('has an image inside the Link', () => {
    const link = wrapper.find('.search-link');
    expect(link.find('.search-link-icon')).toHaveLength(1);
  });
});
