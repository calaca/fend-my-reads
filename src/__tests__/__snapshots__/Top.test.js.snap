// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Top /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Top />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="top-title"
>
          My Reads
</h1>,
        <Link
          className="search-link"
          replace={false}
          to="/search"
>
          <img
                    alt="Search books"
                    className="search-link-icon"
                    src="search.svg"
                    title="Search books"
          />
</Link>,
      ],
      "className": "top",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "My Reads",
          "className": "top-title",
        },
        "ref": null,
        "rendered": "My Reads",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <img
            alt="Search books"
            className="search-link-icon"
            src="search.svg"
            title="Search books"
/>,
          "className": "search-link",
          "replace": false,
          "to": "/search",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Search books",
            "className": "search-link-icon",
            "src": "search.svg",
            "title": "Search books",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="top-title"
>
            My Reads
</h1>,
          <Link
            className="search-link"
            replace={false}
            to="/search"
>
            <img
                        alt="Search books"
                        className="search-link-icon"
                        src="search.svg"
                        title="Search books"
            />
</Link>,
        ],
        "className": "top",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My Reads",
            "className": "top-title",
          },
          "ref": null,
          "rendered": "My Reads",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              alt="Search books"
              className="search-link-icon"
              src="search.svg"
              title="Search books"
/>,
            "className": "search-link",
            "replace": false,
            "to": "/search",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Search books",
              "className": "search-link-icon",
              "src": "search.svg",
              "title": "Search books",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
